src/hooks/useWatchMinedTransactions.ts

```diff
@@ -0,0 +1,156 @@
+import { useCallback } from 'react';
+import { MinedTransaction } from '@/entities';
+import { RainbowError, logger } from '@/logger';
+import { invalidateAddressNftsQueries } from '@/resources/nfts';
+import { userAssetsStore } from '@/state/assets/userAssets';
+import { getPlatformClient } from '@/resources/platform/client';
+import { GetAssetsResponse } from '@/state/assets/types';
+import { time } from '@/utils/time';
+import { getUniqueId } from '@/utils/ethereumUtils';
+import { usePositionsStore } from '@/state/positions/positions';
+import { useClaimablesStore } from '@/state/claimables/claimables';
+import { analytics } from '@/analytics';
+import { event } from '@/analytics/event';
+import { useMinedTransactionsStore } from '@/state/minedTransactions/minedTransactions';
+import { userAssetsStoreManager } from '@/state/assets/userAssetsStoreManager';
+import { convertAmountToRawAmount, greaterThan } from '@/helpers/utilities';
+import { toUnixTime } from '@/worklets/dates';
+import { setUserAssets } from '@/state/assets/utils';
+import { staleBalancesStore } from '@/state/staleBalances';
+
+const ASSET_DETECTION_TIMEOUT = toUnixTime(time.seconds(30));
+
+async function refetchOtherAssets({ address }: { address: string }) {
+  await Promise.all([
+    usePositionsStore.getState().fetch(undefined, { force: true }),
+    useClaimablesStore.getState().fetch(undefined, { force: true }),
+    invalidateAddressNftsQueries(address),
+  ]);
+}
+
+export const useWatchMinedTransactions = ({ address }: { address: string }) => {
+  const nativeCurrency = userAssetsStoreManager(state => state.currency);
+  const clearMinedTransactions = useMinedTransactionsStore(state => state.clearMinedTransactions);
+  const allStaleBalances = staleBalancesStore(state => state.staleBalances);
```

Comment:
Nit: we don't seem to be using `allStaleBalances` should we just do `const staleBalances = staleBalancesStore(state => state.staleBalances[address]);`?